### PR TITLE
rewrite storage account to new generic resource way (WIP)

### PR DESCRIPTION
**Pull request by srb3**
Signed-off-by: srb3 <sbrown@chef.io>

### Description

converts the storage account resources to the azure generic resource
This is WIP as it is currently blocked by at least https://github.com/inspec/inspec-azure/pull/323

### CODE REVIEW

**Review by rmoles — COMMENTED** [review]
Thanks @srb3, this is great. I think all that is required is documentation.

File: docs/resources/azure_storage_account.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+## Syntax
+
+An `azure_storage_account` resource block identifies an Azure storage account by `name` and `resource_group` or the `resource_id`.
+```ruby
+describe azure_storage_account(resource_group: 'rg', name: 'sa')  do
+  it { should exist }
+end
+```
+```ruby
+describe azure_aks_cluster(resource_id: '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.ContainerService/managedClusters/{ClusterName}') do
```
Comment (rmoles) [inline docs/resources/azure_storage_account.md:36]: `azure_aks_cluster` --> `azure_storage_account `

Comment (rmoles) [reply docs/resources/azure_storage_account.md:36]: the resource_id should also be updated

**Review by rmoles — APPROVED** [review]
Thanks @srb3 & @omerdemirok this looks good to go 🎉

File: docs/resources/azure_storage_account.md (hunk 1 above)
Comment (omerdemirok) [reply docs/resources/azure_storage_account.md:36]: Thanks, @rmoles!
Fixed now.

**Review by omerdemirok — APPROVED** [review]
Thanks, @srb3!

**Pull request merged**